src/network/server.rs

```diff
@@ -692,7 +693,8 @@ impl HubService for MyHubService {
                             Ok(hub_event) => {
                                 if filtered_events.contains(&hub_event.r#type) {
                                     let hub_event = Self::rewrite_hub_event(hub_event);
-                                    match tx.send(Ok(hub_event)).await {
+                                    let response = HubEvent::create_hub_event_response(hub_event);
```

Comment:
Let's just use the `rewrite_hub_event` above to add the block number

Reply:
Addressed